godtools/Managers/DownloadResources/DownloadedResourceManager.swift

```diff
@@ -52,79 +52,113 @@ class DownloadedResourceManager: GTDataManager {
             }
     }
     
-    private func saveToDisk(_ resources: [DownloadedResourceJson]) {
-        for remoteResource in resources {
-            let cachedResource = findFirstOrCreateEntity(DownloadedResource.self,
-                                                         byAttribute: "remoteId",
-                                                         withValue: remoteResource.id!)
-            
-            cachedResource.code = remoteResource.abbreviation
-            cachedResource.name = remoteResource.name
-            cachedResource.copyrightDescription = remoteResource.copyrightDescription
-            cachedResource.bannerRemoteId = remoteResource.bannerId
-            cachedResource.totalViews = remoteResource.totalViews!.int32Value
-            
-            for remoteAttachment in (remoteResource.attachments!) {
-                let remoteAttachment = remoteAttachment as! AttachmentResource
-                let cachedAttachment = findFirstOrCreateEntity(Attachment.self,
-                                                               byAttribute: "remoteId",
-                                                               withValue: remoteAttachment.id!)
-                
-                cachedAttachment.sha = remoteAttachment.sha256
-                cachedAttachment.resource = cachedResource
+    func download(_ resource: DownloadedResource) {
+        do {
+            try realm.write {
+                resource.shouldDownload = true
+                TranslationZipImporter.shared.download(resource: resource)
             }
+        } catch {
             
```

Comment:
This is for later as you said, it would be nice to have something generic for dealing with these, and allow to pass a block for the specific actions to do.